src/nvim/lua/executor.c

```diff
@@ -2177,3 +2177,14 @@ char *nlua_funcref_str(LuaRef ref)
   kv_printf(str, "<Lua %d>", ref);
   return str.items;
 }
+
+const char *nlua_read_secure(const char *path)
+{
+  lua_State *const lstate = global_lstate;
+  lua_getglobal(lstate, "vim");
+  lua_getfield(lstate, -1, "readsecure");
+  lua_pushstring(lstate, path);
+  lua_newtable(lstate);
+  lua_call(lstate, 2, 1);
+  return lua_tostring(lstate, -1);
```

Comment:
`vim` and the returned string is not popped from the stack. Popping the string from the stack could free the string, so I think it just has to be copied. `lua_tolstring` will also return the length, that can be used for malloc and memcpy

Reply:
Fixed in 4a2674d4effcdb7a742926000cc9de6d3522c3f0